### PR TITLE
Swallow abort errors in search

### LINKED ISSUE #5639: Unhandled TypeError while using Grand Search
#### Summary

image

#### Expected vs Current Behavior

Expected: no unhandled TypeErrors

#### Steps to Reproduce

1. Type in a search term to the grand search
2. Before the search has completed, delete the term character-by-character
3. Notice errors thrown

#### Environment

Seen during Testathon 8/9/22

```
Version: 2.0.7
Build Date: Thu Aug 04 2022 12:33:07 GMT-0700 (Pacific Daylight Time)
Revision: 15a0a872512556792cac6a8f8cd201a350222312
Branch: release/2.0.7
```

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #5639 #5224

### Describe your changes:
If we have an abort error, swallow it.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by scottbell** [comment]
@ozyx @unlikelyzero  I'm not sure how to add a test for this! I think we'll need:

- CouchDB (search results need to be over network)
- Degraded network ([I think this is possible](https://playwright.dev/docs/network))

I can ruminate a bit more on this, but if you all have any idea, I'm all 👂- thank you!

**Comment by unlikelyzero** [comment]
@scottbell luckily we just talked about this pattern in dev Day yesterday.

I'll reach out to you on slack to get you going

File: e2e/tests/functional/search.e2e.spec.js [hunk 1]
```diff
@@ -92,6 +92,7 @@ test.describe('Grand Search', () => {
 });
 
 test.describe("Search Tests @unstable", () => {
+    test.use({ failOnConsoleError: true });
```
Comment (unlikelyzero) [inline e2e/tests/functional/search.e2e.spec.js:95]: why is this necessary?

File: src/ui/layout/search/GrandSearch.vue [hunk 2]
```diff
@@ -95,6 +95,11 @@ export default {
         },
         getPathsForObjects(objectsNeedingPaths) {
             return Promise.all(objectsNeedingPaths.map(async (domainObject) => {
+                if (!domainObject) {
+                    // user interrupted search, return back
+                    return null;
```
Comment (unlikelyzero) [inline src/ui/layout/search/GrandSearch.vue:100]: can we add coverage for this negative test?

File: src/ui/router/ApplicationRouterSpec.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ describe('Application router utility functions', () => {
             };
 
             openmct.router.on('change:hash', resolveFunction);
+            // We have a debounce set to 300ms on setHash, so if we don't flush,
+            // the above resolve function sometimes doesn't fire due to a race condition.
+            openmct.router.setHash.flush();
```
Comment (unlikelyzero) [inline src/ui/router/ApplicationRouterSpec.js:33]: wow, this is great. I'll add the flaky test closeout to this PR

Comment (scottbell) [reply src/ui/router/ApplicationRouterSpec.js:33]: Hopefully it flakes less! I hate race conditions

File: e2e/tests/functional/search.e2e.spec.js (hunk 1 above)
Comment (scottbell) [reply e2e/tests/functional/search.e2e.spec.js:95]: The issue the PR is addressing was an error being thrown to console, so by enabling this, we can catch this.

File: src/ui/layout/search/GrandSearch.vue (hunk 2 above)
Comment (scottbell) [reply src/ui/layout/search/GrandSearch.vue:100]: I think this is being caught be our search/couch e2e test now, right @ozyx?

Comment (unlikelyzero) [reply src/ui/layout/search/GrandSearch.vue:100]: That's a good catch. We will need to come up with a way to include that suite in our line coverage. I've put a note in our larger, follow-up couchdb testing issue here https://github.com/nasa/openmct/issues/5686

File: e2e/tests/functional/search.e2e.spec.js (hunk 1 above)
Comment (unlikelyzero) [reply e2e/tests/functional/search.e2e.spec.js:95]: Oh, the failOnConsoleError should be set to true by default

Comment (scottbell) [reply e2e/tests/functional/search.e2e.spec.js:95]: Ah! That makes sense. I'll remove it.

File: src/ui/layout/search/GrandSearch.vue (hunk 2 above)
Comment (ozyx) [reply src/ui/layout/search/GrandSearch.vue:100]: > I think this is being caught be our search/couch e2e test now, right @ozyx?

Yes, it should be. Might not have been registered as being covered because we didn't trigger the couchdb e2e tests to run against this PR?

Comment (scottbell) [reply src/ui/layout/search/GrandSearch.vue:100]: Yeah, if it didn’t run, probably didn’t accumulate code coverage.

File: src/ui/layout/search/GrandSearch.vue [hunk 4]
```diff
@@ -127,12 +132,17 @@ export default {
...
-                console.error(`😞 Error searching`, error);
                 this.searchLoading = false;
 
                 if (this.abortSearchController) {
                     delete this.abortSearchController;
                 }
+
+                // Is this coming from the AbortController?
+                // If so, we can swallow the error. If not, 🤮 it to console
+                if (!error.name === 'AbortError') {
```
Comment (ozyx) [inline src/ui/layout/search/GrandSearch.vue:143]: Is the `!error.name` intentional or a typo ?

**Review by ozyx — COMMENTED** [review]
I have one question for you, otherwise this is good to go.

Double-checking your PR led to the discovery that our CouchDB CI workflow was not functioning correctly. After finally emerging from the rabbit hole, now it is! So that's exciting 😄

File: src/ui/layout/search/GrandSearch.vue (hunk 4 above)
Comment (scottbell) [reply src/ui/layout/search/GrandSearch.vue]: typo! nice catch

**Review by ozyx — APPROVED** [review]
Let's GO! 🚀
